### DISCUSSION TITLE
how can use WebSocket in axmol?

### DISCUSSION THREAD

**Discussion by armanhossiny** [axmolengine/axmol#1468] [General]
how can use websocket in axmol?

**Comment by rh101** [comment]
> how can use websocket in axmol?

You can use any websockets library with Axmol. 

The one that was used in Cocos2d-x is libwebsockets. You can find more info about it here: https://libwebsockets.org/

This is another one you can use: https://github.com/machinezone/IXWebSocket
An example of how you can link IXWebSocket to your project:

```
set(USE_TLS TRUE CACHE BOOL "Enable TLS support" FORCE)
set(USE_OPEN_SSL ON)
set(OPENSSL_FOUND TRUE)
set(OPENSSL_LIBRARIES OpenSSL::Crypto OpenSSL::SSL)
add_subdirectory(${CMAKE_CURRENT_SOURCE_DIR}/thirdparty/IXWebSocket ${APP_THIRDPARTY_OUTPUT_DIR}/IXWebSocket)
get_target_property(ixwebsocket_INCLUDE_DIRS ixwebsocket INTERFACE_INCLUDE_DIRECTORIES)
target_link_libraries(${APP_NAME} ixwebsocket)
target_include_directories(${APP_NAME}
    PRIVATE ${ixwebsocket_INCLUDE_DIRS}
)
```
This assumes IXWebSocket is in the `[your_project]/thirdparty/IXWebSocket` folder.

**Reply by armanhossiny** [reply]
Thank you :)

Why does it not have a websocket by default like Cocos2dx?

Is it possible to add it in the future?

**Reply by armanhossiny** [reply]
hi dear rh01
I add library working for android and desktop but not working for web and show error

https://libwebsockets.org/

**Reply by rh101** [reply]
> I add library working for android and desktop but not working for web and show error

I'm glad you got it working, but I'm sorry I can't help you with those issues. Perhaps you should ask the developers of libwebsockets.

**Reply by armanhossiny** [reply]
thank you so much
